ui/src/components/common/CustomAlert/CustomAlert.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState, useEffect } from 'react';
+import './CustomAlert.css';
+import PropTypes from 'prop-types';
+import { Alert, AlertActionCloseButton } from '@patternfly/react-core';
+
+/**
+ * A custom alert component that wraps the Patternfly 4 Alert.
+ *
+ * @param {Object} visible - The visible state of the alert.
+ * @param {func} onClose - The callback function to execute on close.
+ * @param {*} children - Any child props - usually the description.
+ * @param {Object} props - The component props
+ */
+const CustomAlert = ({ visible, onClose, children, ...props }) => {
+  const [isVisible, setVisibility] = useState(visible);
+
+  useEffect(() => {
+    setVisibility(visible);
+  }, [visible]);
+
+  if (!isVisible) {
+    return null;
+  }
+
+  return (
+    <Alert
+      className='Alert'
+      {...props}
+      action={<AlertActionCloseButton onClose={onClose} />}>
+      {children}
+    </Alert>
+  );
+};
+
+CustomAlert.propTypes = {
+  visible: PropTypes.bool.isRequired,
+  onClose: PropTypes.func,
```

Comment:
Do you think this should be a required proptype?

Reply:
Yes it should be.